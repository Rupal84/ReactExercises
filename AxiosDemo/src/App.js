import React from 'react';
import './App.css';
import AppComp from './AppComp';

function App() {
  return (
    <div className="App">
      <AppComp/>
    </div>
  );
}

export default App;
