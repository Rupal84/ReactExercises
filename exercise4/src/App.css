.App {
  display: flex;
  flex-direction: row;
  justify-content: space-between;
  max-width: 500px;
}
.card {
  height: 220px;
  width: 200px;
  border: 1px #4d4d4d solid;
}
p {
  margin-block-start: 0%;
  margin-block-end: 0%;
}