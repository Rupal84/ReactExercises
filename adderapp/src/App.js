import React from 'react';
import './App.css';

class App extends React.Component {
  
  state = {
    value: 1
  }
  render() {
    return (
      <div className="App">
       <button className='addButton' onClick={this.doAddition}>Add</button>
       <p>Added value:{this.state.value}</p>
      </div>
    );
  }
    
   doAddition = ()=>{
    var val = this.state.value;
    this.setState({value: val + val})
  }
  
}

export default App;
