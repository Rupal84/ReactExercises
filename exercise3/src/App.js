import React from 'react';
import './App.css';

class App extends React.Component {
   state = {}
  constructor() {
    super();
   this.state = {
      show: true
    }
  }
  render() {
  return (
    <div className="App">
      <button className="ui primary button"
        onClick={()=>this.setState({show: true})}>
          Show text
      </button>
      {this.state.show?this.sayHello():<br/>}
      <button  className="ui inverted primary button" onClick={()=>this.setState({show: false})}>Hide text</button>
    </div>
  );
  }

   sayHello = ()=> {
      return (
        <p className="ui ">Hello React</p>
      )
  }
}

export default App;
