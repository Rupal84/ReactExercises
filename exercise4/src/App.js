import React from 'react';
import Mobile from './Mobile';
import './App.css';

function App() {
  return (
    <div className="App">
      {getMobiles()}
    </div>
  );
}

var phones = [
  {
      model : 'Google Pixel 2 XL',
      price: '62000',
      ram: '4GB',
      camera: '12.2MP',
      os: 'Android pie',
      processor: 'Snapdragon 835 octa core'
  },
  {
    model : 'Apple Iphone X',
    price: '98000',
    ram: '3GB',
    camera: '12MP',
    os: 'iOS v11.1.1',
    processor: 'A11 Bionic hexa-core'
}
]

function getMobiles()
{
  return (
    phones.map((phone)=> {
      return (
        <Mobile phone={phone}></Mobile>
      )
    })
  )
}
export default App;
