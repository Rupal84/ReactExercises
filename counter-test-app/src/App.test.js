import React from "react";
import App, { Counter } from "./App";
import renderer from 'react-test-renderer';
import { shallow } from "enzyme";
describe("App Component", () => {
  test("renders the Counter wrapper", () => {
    const wrapper = shallow(<App />);
    expect(wrapper.find(Counter).length).toEqual(1);
  });
  test("passes all props to Counter wrapper", () => {
    const wrapper = shallow(<App />);
    let counterWrapper = wrapper.find(Counter);
    expect(counterWrapper.props().counter).toEqual(0);
    wrapper.setState({ counter: -1 });
    counterWrapper = wrapper.find(Counter);
    expect(counterWrapper.props().counter).toEqual(-1);
  });
  test("increments the counter", () => {
    const wrapper = shallow(<App />);
    wrapper.setState({ counter: 0 });
    wrapper.find("button.increment").simulate("click");
    expect(wrapper.state().counter).toEqual(1);
  });
  test("decrements the counter", () => {
    const wrapper = shallow(<App />);
    wrapper.setState({ counter: 0 });
    wrapper.find("button.decrement").simulate("click");
    expect(wrapper.state().counter).toEqual(-1);
  });
});
describe("App Snapshot", () => {
  test("renders", () => {
    const component = renderer.create(<App />);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
describe("Counter Snapshot", () => {
  test("renders", () => {
    const component = renderer.create(<Counter counter={1} />);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});


