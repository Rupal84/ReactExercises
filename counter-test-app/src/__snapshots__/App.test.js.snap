// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Snapshot renders 1`] = `
<div>
  <h4>
    My Counter
  </h4>
  <p>
    0
  </p>
  <button
    className="increment"
    onClick={[Function]}
    type="button"
  >
    Increment
  </button>
           
  <button
    className="decrement"
    onClick={[Function]}
    type="button"
  >
    Decrement
  </button>
</div>
`;

exports[`Counter Snapshot renders 1`] = `
<p>
  1
</p>
`;
